core/services/bulletprooftxmanager/bulletprooftxmanager.go

```diff
@@ -146,7 +156,7 @@ func (b *BulletproofTxManager) Start() (merr error) {
 		logger.Debugw("BulletproofTxManager: booting", "keys", keys)
 
 		eb := NewEthBroadcaster(b.db, b.ethClient, b.config, b.keyStore, b.advisoryLocker, b.eventBroadcaster, keys, b.gasEstimator)
-		ec := NewEthConfirmer(b.db, b.ethClient, b.config, b.keyStore, b.advisoryLocker, keys, b.gasEstimator)
+		ec := NewEthConfirmer(b.db, b.ethClient, b.config, b.keyStore, b.advisoryLocker, keys, b.gasEstimator, b.resumeCallback)
```

Comment:
Register dynamically instead

Reply:
We can't do this because the EthConfirmer is only created after `Start` is called and it also gets recreated any time `<-keysChanged`. The toplevel struct needs to track the callbacks, or we need to copy them over when recreating the `ec`.

Reply:
I think you can keep them inside the confirmer object and just expose a CopyWithNewKeys(newKeys) function or something to do a ```b.ec = b.ec.CopyWithNewKeys(newKeys)```

Reply:
We could have an assert that `b.resumeCallback` is not nil here - it's supposed to be always set when Start() is called